.circleci/config.yml

```diff
@@ -69,28 +69,19 @@ references:
         # Install cmake >=3.14 to support Visual Studio 2019 generator
         python${PYVER} -m pip install --progress-bar off -U cmake~=3.14
 
-  get_style_requirements: &get_style_requirements
+  get_pre_commit_requirements: &get_pre_commit_requirements
     run:
-      name: Install style check dependencies
+      name: Install pre-commit dependencies
       command: |
-        python${PYVER} -m pip install --user flake8==3.8.3
-
-  #check_cpp_style: &check_cpp_style
-  #  run:
-  #    name: clang-format
-  #    command: |
-  #      for f in cpp/**/*; do
-  #        if ! diff $f <(clang-format-6.0 -style=file $f) > /dev/null; then
-  #          echo "$f does not match clang-format style."
-  #          diff --color $f <(clang-format-6.0 -style=file $f) || true
-  #        fi
-  #      done
-
-  check_python_style: &check_python_style
+        python${PYVER} -m pip install --user -r requirements-precommit.txt
+
+  pre-test-checks: &pre-test-checks
     run:
-      name: flake8
+      name: Run all static checks
       command: |
-        python${PYVER} -m flake8 --show-source .
+        # Modifying path in previous steps doesn't seem to hold, so specifying
+        # the full path explicitly here seems cleaner.
+        "`python${PYVER} -m site --user-base`/bin/pre-commit" run --all-files
```

Comment:
Changes to environment variables are not persistent between "run" steps in CircleCI. This might also be fixed by adding `conda activate base` to this step, which should alter the path.

Reply:
We only use conda for the Windows build :P but yes, I forgot about that until after I tried this. I can make a note to that effect here.

Reply:
Oh, right, ignore what I said about conda. However, it might possible to use `python -m pre-commit` instead of the path hack?

Reply:
I'm dumb. I had tried exactly that and it failed and I was too lazy to figure out why. Of course, it's because python package names don't have hyphens; `python -m pre_commit` works.